pennylane/transforms/compile.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Code for the high-level quantum function transform that executes compilation."""
+
+from inspect import getmembers, isclass
+
+from pennylane import apply
+from pennylane.tape import get_active_tape
+
+from pennylane.transforms import qfunc_transform
+from pennylane.transforms.optimization import cancel_inverses, commute_controlled, merge_rotations
+
+from pennylane import templates
+
+# Get a list of the names of existing templates in PennyLane
+template_list = [cls[0] for cls in getmembers(templates, isclass)]
```

Comment:
😬 

There must be a better solution to this? I can't think of one of the top of my head though

Reply:
Me neither :disappointed: I've been stuck on this since the original prototype.